frontend/main.py

```diff
@@ -316,11 +320,11 @@ def send_email(
     )
 
 
-def save_correction_to_hadith_table(urn: int, corrected_val: str):
+def save_correction_to_hadith_table(urn: int, corrected_val: str, attribute: str):
     conn = pymysql.connect(**MYSQL_PROPS)
     cursor = conn.cursor()
-    query = "UPDATE bukhari_english SET hadithText = %(hadith_text)s WHERE englishURN = %(urn)s;"
-    cursor.execute(query, {"hadith_text": corrected_val, "urn": urn})
+    query = str("UPDATE bukhari_english SET ") + attribute + str(" = %(corrected_val)s WHERE englishURN = %(urn)s;")
```

Comment:
can you please use a formatted string instead of concatenated strings?
Please validate the attribute name against a list of allowed attribute names to avoid SQL injection

Reply:
updated, kindly to review again 🙏🏽